crates/matrix-sdk-ui/src/room_list_service/room.rs

```diff
@@ -83,15 +83,9 @@ impl Room {
         self.inner.room.room_id()
     }
 
-    /// Get the best possible name for the room.
-    ///
-    /// If the sliding sync room has received a name from the server, then use
-    /// it, otherwise, let's calculate a name.
+    /// Get the name of the room if it exists.
     pub async fn name(&self) -> Option<String> {
-        Some(match self.inner.sliding_sync_room.name() {
-            Some(name) => name,
-            None => self.inner.room.display_name().await.ok()?.to_string(),
-        })
+        Some(self.inner.room.display_name().await.ok()?.to_string())
```

Comment:
Are we certain that this is not needed? The proxy calculates the room name for us even before we have all the members or events, i.e. it does what `display_name()` does for us.

I see that we're storing whatever the proxy sends to us as the `m.room.name` event in the `RoomInfo` since https://github.com/matrix-org/matrix-rust-sdk/pull/2032.

But that PR seems to be questionable at best. Though the PR should mean that these two method do contain the same info.

Reply:
Oops, auto-merge strikes again :fearful:

Reply:
The SS room name is synced with the room and room info during the sync process. So yeah, this is not needed to keep it in SS room anymore.

Reply:
And to reply concretely to your mesasge, normally `display_name` returns early if `RoomInfo` contains a name, https://github.com/matrix-org/matrix-rust-sdk/blob/40e006658cfee2d56e7b3e36d97fc9a78de08295/crates/matrix-sdk-base/src/rooms/normal.rs#L568-L575.

Also, we have this test that ensures the behaviour at the sliding sync level, https://github.com/matrix-org/matrix-rust-sdk/blob/40e006658cfee2d56e7b3e36d97fc9a78de08295/crates/matrix-sdk-base/src/sliding_sync.rs#L1212-L1232.

Reply:
> And to reply concretely to your mesasge, normally `display_name` returns early if `RoomInfo` contains a name,

I know what `display_name()` does, it follows the spec: https://spec.matrix.org/unstable/client-server-api/#calculating-the-display-name-for-a-room.

What I'm complaining about is that we put something that isn't a `m.room.name` event content into the field that always should contain a `m.room.name` event content.

The thing that the SS proxy returns **is not** a `m.room.name` event, it's the result of the calculation I linked previously done by the proxy.

Incidentally this thing works, because we created a fake `m.room.name` event and put it in the same field, the spec tells you to take this field verbatim. The problem is that this data, while also being a string, is semantically very different.

This might now contain the field "Tom, Dick, and Harry" for a three person room, but this string should be localized and we lost any way of doing and detecting this case because we pretended that it's actually an `m.room.name` event that contains "Tom, Dick, and Harry".

I should probably post this complaint on the PR instead or just open a new issue. I find such hacks that make things quickly work, but mangle with the correctness and make future features harder to implement, for example like the localization of room display names, the bane of our existence.